resources/contribute-documentation/tutorial-guidelines.md

```diff
@@ -1,84 +1,129 @@
 
 
-## Use Cases
+### Dependencies
 
-Use cases describe how to pull together multiple features to create a practical application that solves a business problem. They provide context and assistance with the decision making process, then provide links to the appropriate topics for each step of implementation.
+Dependencies can be a source of maintenance burden, because you need to stay up-to-date with security fixes and breaking changes to the dependencies. On the other hand, reimplementing common utilities in every code sample is its own maintenance burden, and it's even worse to "roll your own" security-sensitive code. Some users may be working on codebases that are locked into competing/incompatible dependencies, which can make it harder to adapt your code to their situation; the more dependencies you have, the more likely this is to occur. 
 
+Some guidelines:
 
-# Tutorial Components
+1. Prefer standard library functions to third-party libraries, even if they're not quite as convenient to use.
+    - Use third-party libraries when they're _significantly_ more convenient than the standard library. For example, [even Python's official documentation recommends using the Requests lib instead of `urllib.request`](https://docs.python.org/3/library/urllib.request.html#module-urllib.request).
+    - When updating old code samples, look for cases where dependencies can be eliminated because the standard library has grown to encompass functionality that previously needed a library.
+2. Implement your own functions when they're small and not security-sensitive; use libraries for complex or security-sensitive functions.
+3. Prefer widely-used, actively maintained libraries.
 
-This section describes the elements of the modular tutorials used on XRPL.org.
+### General Guidelines and Best Practices
 
+The following guidelines apply for XRP Ledger code samples regardless of language:
 
-## Sample Application
+- Don't hardcode secret keys, even example keys that don't hold real money. Instead, do any of the following:
+    - Fund a new account using the faucet.
+    - Prompt the user to paste the seed of the account they want to use.
+- Use `client` as the name for the API client instance.
+- Print output to the console, especially before doing any network operations such as calling API methods or submitting transactions.
+- Use the client library's "submit and wait" function when sending transactions. Autofill, sign, and submit the transaction all in one call.
+- Use tabs for code samples even if you only have a code sample in one language (yet).
+- When making WebSocket/JSON-RPC API calls, use the latest API version and the `validated` ledger.
 
-XRPL tutorial code samples are modular in nature. For example, Script 1 demonstrates how to create a test account, access the XRP Ledger, and transfer XRP between accounts. Any further samples can reuse the functions in Script 1.
+### Language-specific Guidelines
```

Comment:
Question for our language agnostic tutorials specifically. If we provide the WebSocket/JSON-RPC API call tab, should the WebSocket/JSON-RPC call always include the expected response?

For an example, see [Send a Time-Held Escrow](https://xrpl.org/docs/tutorials/how-tos/use-specialized-payment-types/use-escrows/send-a-time-held-escrow#2.-submit-escrowcreate-transaction).

Would be good to clarify here.